app/services/record_declaration.rb

```diff
@@ -81,14 +82,13 @@ def milestone
     schedule.milestones.find_by(declaration_type:)
   end
 
-  def schedule
-    participant_profile&.schedule
-  end
-
 private
 
   attr_writer :raw_declaration_date
 
+  delegate :schedule, to: :participant_profile, allow_nil: true
+  delegate :cohort, to: :schedule
```

Comment:
I think it's best to get cohort from the induction record rather than profile -> schedule -> cohort incase there might be inconsistent data